src/gov/nist/javax/sip/stack/IOHandler.java

```diff
@@ -106,9 +106,9 @@ protected Socket getSocket(String key) {
 
     protected void removeSocket(String key) {
         socketTable.remove(key);
-        if ( socketCreationMap.get(key) != null ) {
-        	socketCreationMap.get(key).release();
-        	socketCreationMap.remove(key);
+        Semaphore s = socketCreationMap.remove(key);
```

Comment:
this should be safe, although performance penalty is hard to anticipate, better to be safe by now :+1